unlock-app/src/components/interface/keychain/Key.tsx

```diff
@@ -25,13 +26,31 @@ import { AuthenticationContext } from '../../../contexts/AuthenticationContext'
 import { MAX_UINT } from '../../../constants'
 import { ConfigContext } from '../../../utils/withConfig'
 import { OpenSeaIcon } from '../../icons'
+import { CancelAndRefundModal } from './CancelAndRefundModal'
 
 interface KeyBoxProps {
   tokenURI: string
   lock: any
   expiration: string
   keyId: string
   network: number
+  isKeyExpired: boolean
+  expirationStatus: string
+}
+
+function ExpiredTag() {
+  return (
+    <p className="bg-red-100 border inline-flex items-center gap-2 border-brand-secondary text-brand-secondary text-xs font-semibold mr-2 px-2.5 py-0.5 rounded-full">
+      <DangerIcon /> Expired
+    </p>
+  )
+}
+function ValidTag() {
+  return (
+    <p className="bg-green-100 border inline-flex items-center gap-2 border-green-600 text-green-600 text-xs font-semibold mr-2 px-2.5 py-0.5 rounded-full">
+      <CheckIcon /> Valid
+    </p>
+  )
 }
 
```

Comment:
This is overriding the latest code on the master branch. We should be using Badge component.